File: src/main/java/com/paragon/client/systems/module/impl/client/DiscordRPC.java
package com.paragon.client.systems.module.impl.client;

import com.paragon.Paragon;
import com.paragon.client.systems.module.IgnoredByNotifications;
import com.paragon.client.systems.module.Module;
import com.paragon.client.systems.module.Category;

@IgnoredByNotifications
public class DiscordRPC extends Module {

    public DiscordRPC() {
        super("DiscordRPC", Category.CLIENT, "Changes your Discord presence to reflect the client's current state");
    }

    @Override
    public void onEnable() {
        Paragon.INSTANCE.getPresenceManager().startRPC();
    }

    @Override
    public void onTick() {
        Paragon.INSTANCE.getPresenceManager().updateRPC();
    }

    @Override
    public void onDisable() {
        Paragon.INSTANCE.getPresenceManager().stopRPC();
    }

}

File: src/main/java/com/paragon/client/systems/module/impl/client/DiscordRPC.kt
package com.paragon.client.systems.module.impl.client

import com.paragon.Paragon
import com.paragon.api.module.Category
import com.paragon.api.module.IgnoredByNotifications
import com.paragon.api.module.Module
import com.paragon.api.setting.Setting

/**
 * @author Surge
 */
@IgnoredByNotifications
object DiscordRPC : Module("DiscordRPC", Category.CLIENT, "Changes your Discord presence to reflect the client's current state") {

    val showServer = Setting(
        "Show Server",
        true
    ) describedBy "Show the servers ip in the RPC"

    override fun onEnable() {
        Paragon.INSTANCE.presenceManager.startRPC()
    }

    override fun onTick() {
        Paragon.INSTANCE.presenceManager.updateRPC()
    }

    override fun onDisable() {
        Paragon.INSTANCE.presenceManager.stopRPC()
    }

}